recipes/agent_build_tools.rb

```diff
@@ -52,8 +52,43 @@
   only_if { has_nvidia_support? }
 end
 
-# GeForce GTX 550 Ti requires old 3xx.xx series
-package 'nvidia-384' do
+if has_nvidia_support? and nvidia_devices.size != 1
+  Chef::Log.warn("There are multiple nvidia devices and I am only looking at the first!")
+end
+nvidia_device = if has_nvidia_support?
+                  nvidia_devices.first['device']
+                else
+                  nil
+                end
+
+nvidia_driver = case nvidia_device
+                when /GTX 550/
+                  # Old optimus machine in OSRF office
+                  'nvidia-384'
+                when /GRID K520/
+                  # AWS g2 instances
+                  'nvidia-384'
+                when /Tesla M60/
+                  # AWS g3 instances
+                  'nvidia-driver-470'
+                when /Tesla T4/
+                  # AWS g4dn instances
+                  'nvidia-driver-470'
+                when /Device 2237/
+                  # AWS g5 instances
+                  'nvidia-driver-470'
+                when nil
+                  # it doesn't really matter which driver we use
+                  # if there is no nvidia device but specify
+                  # one to avoid a warning.
+                  'nvidia-driver-470'
+                else
+                  Chef::Log.warn("Untested GPU `#{nvidia_devices.first['device']}` being used. Assuming a functioning driver")
+                  'nvidia-driver-470'
+                end
```

Comment:
If this grows much more it may make sense to move it into a helper method, especially if different considerations like platform start being considered as well.